### ISSUE TITLE
[BUG] Mining on single RTX3070 stats show 0 shares, pools show shares

### ISSUE THREAD

**Issue by RainbowMiner** [sp-hash/TeamBlackMiner#14]
Hi,

there seems to be a bug with the stats. I have one of the mixed GPU rig (6xGTX1070, 1xRTX3070) mining on that single RTX3070 with v1.02 release, but it doesn't report any shares. No accepted, no rejected, no stale. But looking at Ethermine pool stats, the miner in fact does find shares. Maybe one note: there are running two instances of TBminer.exe, currently. One mines on the six GTX1070, the other runs on the RTX3070. The one mining the GTX1070 does show correct stats.

![image](https://user-images.githubusercontent.com/39437538/132997367-d8cd36ed-1099-45e2-b383-df32bc5506ce.png)

Miner log RTX3070:
```
     0m [2021-09-12 19:09:45.601] OpenCL driver version: 27.20
     0m [2021-09-12 19:09:45.601] Cuda driver version: 11.4
     0m [2021-09-12 19:09:45.601] Cuda runtime API version: 11.4
     0m [2021-09-12 19:09:45.601] 
     0m [2021-09-12 19:09:45.601]                                 
     0m [2021-09-12 19:09:45.601] 
     0m [2021-09-12 19:09:45.601] Welcome to Team Black Miner 1.02
     0m [2021-09-12 19:09:45.601]                                 
     0m [2021-09-12 19:09:45.601] 
     0m [2021-09-12 19:09:45.859] GPU6 NVIDIA GeForce RTX 3070 8GB (compute 8.6) added to miner as Cuda device
     0m [2021-09-12 19:09:46.057] Trying eu1.ethermine.org:4444 (ethash)
     0m [2021-09-12 19:09:46.088] Extranonce is set to 3777675662433714176
     0m [2021-09-12 19:09:46.088] Pool responded successfully to subscribe
     0m [2021-09-12 19:09:46.088] difficulty is set to 1.000000
     0m [2021-09-12 19:09:46.135] Pool responded successfully to extranonce subscribe
     0m [2021-09-12 19:09:46.150] Pool responded successfully to authorize
     0m [2021-09-12 19:09:47.171] GPU0 Started Cuda work thread
     0m [2021-09-12 19:09:52.954] GPU6 Starting generating ethash dag (Epoch: 440)
     0m [2021-09-12 19:09:53.064] GPU0 DAG generated in 5.89 seconds (4653054kb)
     0m [2021-09-12 19:09:53.064] GPU0 XIntensity set to Dynamic
...
...
...
    24m [2021-09-12 19:35:03.028] Job 071da8
    24m [2021-09-12 19:35:04.090] Job b5f6ca
    24m [2021-09-12 19:35:07.339] Job acca67
    25m [2021-09-12 19:35:11.104] 
    25m [2021-09-12 19:35:11.104] 
    25m [2021-09-12 19:35:11.104] eu1.ethermine.org:4444 (ethash)    PING: 50ms    DIFFICULTY: 1.00     EPOCH: 440
    25m [2021-09-12 19:35:11.104]                                                                                 
    25m [2021-09-12 19:35:11.104] 
    25m [2021-09-12 19:35:11.104] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
    25m [2021-09-12 19:35:11.104] GPU6	3070	0/0	0	0	0	0	0.00	0.00
    25m [2021-09-12 19:35:11.104] 						0	0.00	0.00
    25m [2021-09-12 19:35:11.104] 
    25m [2021-09-12 19:35:11.104] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
    25m [2021-09-12 19:35:11.104] GPU6	3070	Cuda	0.00 H/W	60.94 MH/s	0.00 H/s	0/0/0 (100.00)
    25m [2021-09-12 19:35:11.104] 			0.00 H/s	60.94 MH/s	0.00 H/s	0/0/0 (100.00)
    25m [2021-09-12 19:35:11.104]                                                                                 
    25m [2021-09-12 19:35:11.104] 
    25m [2021-09-12 19:35:12.588] Job 6b1348
    25m [2021-09-12 19:35:16.759] Job c0546c
```

Miner log six GTX1070:
```
     0m [2021-09-12 19:09:44.702] OpenCL driver version: 27.20
     0m [2021-09-12 19:09:44.702] Cuda driver version: 11.4
     0m [2021-09-12 19:09:44.702] Cuda runtime API version: 11.4
     0m [2021-09-12 19:09:44.702] 
     0m [2021-09-12 19:09:44.702]                                 
     0m [2021-09-12 19:09:44.702] 
     0m [2021-09-12 19:09:44.702] Welcome to Team Black Miner 1.02
     0m [2021-09-12 19:09:44.702]                                 
     0m [2021-09-12 19:09:44.702] 
     0m [2021-09-12 19:09:45.046] GPU0 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-12 19:09:45.195] GPU1 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-12 19:09:45.330] GPU2 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-12 19:09:45.450] GPU3 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-12 19:09:45.559] GPU4 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-12 19:09:45.771] GPU5 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-12 19:09:46.015] Trying eu1.ethermine.org:4444 (ethash)
     0m [2021-09-12 19:09:46.105] Extranonce is set to 3342233873462329344
     0m [2021-09-12 19:09:46.105] Pool responded successfully to subscribe
     0m [2021-09-12 19:09:46.105] difficulty is set to 1.000000
     0m [2021-09-12 19:09:46.150] Pool responded successfully to extranonce subscribe
...
...
    24m [2021-09-12 19:35:11.791] Job 6b1348
    25m [2021-09-12 19:35:14.668] 
    25m [2021-09-12 19:35:14.668] 
    25m [2021-09-12 19:35:14.668] eu1.ethermine.org:4444 (ethash)    PING: 58ms    DIFFICULTY: 1.00     EPOCH: 440
    25m [2021-09-12 19:35:14.668]                                                                                 
    25m [2021-09-12 19:35:14.668] 
    25m [2021-09-12 19:35:14.668] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
    25m [2021-09-12 19:35:14.668] GPU0	1070	62/0	42	2014	4404	135	0.14	inf
    25m [2021-09-12 19:35:14.668] GPU1	1070	66/0	62	1993	4404	132	0.13	inf
    25m [2021-09-12 19:35:14.668] GPU2	1070	67/0	62	1967	4404	130	0.13	inf
    25m [2021-09-12 19:35:14.668] GPU3	1070	67/0	62	1937	4404	134	0.13	inf
    25m [2021-09-12 19:35:14.668] GPU4	1070	63/0	46	2002	4404	133	0.13	inf
    25m [2021-09-12 19:35:14.668] GPU5	1070	64/0	50	1992	4404	129	0.13	inf
    25m [2021-09-12 19:35:14.668] 						793	0.79	inf
    25m [2021-09-12 19:35:14.668] 
    25m [2021-09-12 19:35:14.668] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
    25m [2021-09-12 19:35:14.668] GPU0	1070	Cuda	199.00 kH/W	26.86 MH/s	27.68 MH/s	12/0/6 (100.00)
    25m [2021-09-12 19:35:14.668] GPU1	1070	Cuda	202.47 kH/W	26.73 MH/s	27.34 MH/s	9/0/2 (100.00)
    25m [2021-09-12 19:35:14.668] GPU2	1070	Cuda	204.40 kH/W	26.57 MH/s	27.10 MH/s	9/0/3 (100.00)
    25m [2021-09-12 19:35:14.668] GPU3	1070	Cuda	195.04 kH/W	26.14 MH/s	26.52 MH/s	10/0/2 (100.00)
    25m [2021-09-12 19:35:14.668] GPU4	1070	Cuda	198.39 kH/W	26.39 MH/s	27.27 MH/s	14/0/7 (100.00)
    25m [2021-09-12 19:35:14.668] GPU5	1070	Cuda	204.52 kH/W	26.38 MH/s	27.23 MH/s	10/0/4 (100.00)
    25m [2021-09-12 19:35:14.668] 			1203.81 kH/s	159.07 MH/s	163.13 MH/s	64/0/24 (100.00)
    25m [2021-09-12 19:35:14.668]                                                                                 
    25m [2021-09-12 19:35:14.668] 
    25m [2021-09-12 19:35:15.978] Job c0546c
```

**Comment by RainbowMiner** [comment]
I suspect that's possibly a problem, due to both TBminer.exe processes try to access nvidia-smi. I do see `this process cannot access the file because it is being used by another process` in the logs, from time to time.

**Comment by sp-hash** [comment]
try to run both instances with --no-cpu

**Comment by RainbowMiner** [comment]
> try to run both instances with --no-cpu

Both instances are running with --no-cpu already

**Comment by RainbowMiner** [comment]
I tried with only one instance running on that GPU - the problem persists. From the programmer perspective I'ld strongly suspect an internal indexing problem, since it is CUDA GPU6 (as used in the stats table) that is selected, but TBMiner uses GPU0 internally. That would explain the missing health data (temp, fan, coreclock, ...), too.

```
     0m [2021-09-13 10:07:24.340] GPU6 NVIDIA GeForce RTX 3070 8GB (compute 8.6) added to miner as Cuda device
     0m [2021-09-13 10:07:24.434] Trying daggerhashimoto.eu-west.nicehash.com:3353 (ethash)
     0m [2021-09-13 10:07:24.621] Extranonce is set to 2487110695192100864
     0m [2021-09-13 10:07:24.621] Pool responded successfully to subscribe
     0m [2021-09-13 10:07:24.684] Pool responded successfully to extranonce subscribe
     0m [2021-09-13 10:07:24.699] Pool responded successfully to authorize
     0m [2021-09-13 10:07:24.715] difficulty is set to 0.231964
     0m [2021-09-13 10:07:24.715] GPU0 Started Cuda work thread
     0m [2021-09-13 10:07:30.417] GPU6 Starting generating ethash dag (Epoch: 440)
     0m [2021-09-13 10:07:30.542] GPU0 DAG generated in 5.80 seconds (4653054kb)
     0m [2021-09-13 10:07:30.542] GPU0 XIntensity set to Dynamic
     0m [2021-09-13 10:07:30.542] Job 000000003f275f7f
     0m [2021-09-13 10:07:41.336] Job 000000003f275f81
     0m [2021-09-13 10:07:43.070] GPU0 Dag buffer verified ok!
     0m [2021-09-13 10:07:46.273] Job 000000003f275f82
     0m [2021-09-13 10:07:47.069] Job 000000003f275f83
     0m [2021-09-13 10:07:51.068] Job 000000003f275f84
     0m [2021-09-13 10:07:51.865] Job 000000003f275f85
     0m [2021-09-13 10:07:53.771] GPU0 Solution found in 2.69s (Submit every 30.51 seconds)
     0m [2021-09-13 10:07:55.067] Job 000000003f275f87
     0m [2021-09-13 10:08:01.206] GPU0 Solution found in 6.33s (Submit every 18.98 seconds)
     0m [2021-09-13 10:08:02.316] Job 000000003f275f89
     0m [2021-09-13 10:08:08.830] Job 000000003f275f8a
     0m [2021-09-13 10:08:09.626] Job 000000003f275f8b
     0m [2021-09-13 10:08:16.140] Job 0000000036332f7d
     0m [2021-09-13 10:08:19.999] GPU0 Solution found in 4.60s (Submit every 18.91 seconds)
     0m [2021-09-13 10:08:22.483] Job 0000000036332f7e
     0m [2021-09-13 10:08:23.607] Job 0000000036332f7f
     0m [2021-09-13 10:08:24.654] Job 0000000036332f80
     1m [2021-09-13 10:08:25.154] 
     1m [2021-09-13 10:08:25.154] 
     1m [2021-09-13 10:08:25.154] daggerhashimoto.eu-west.nicehash.com:3353 (ethash)    PING: 54ms    DIFFICULTY: 0.232    EPOCH: 440
     1m [2021-09-13 10:08:25.154]                                                                                                    
     1m [2021-09-13 10:08:25.154] 
     1m [2021-09-13 10:08:25.154] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     1m [2021-09-13 10:08:25.154] GPU6	3070	0/0	0	0	0	0	0.00	0.00
     1m [2021-09-13 10:08:25.154] 						0	0.00	0.00
     1m [2021-09-13 10:08:25.154] 
     1m [2021-09-13 10:08:25.154] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     1m [2021-09-13 10:08:25.154] GPU6	3070	Cuda	0.00 H/W	63.98 MH/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 10:08:25.154] 			0.00 H/s	63.98 MH/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 10:08:25.154]

**Comment by sp-hash** [comment]
good point. We will check it out

**Comment by sp-hash** [comment]
Please try v1.04

**Comment by RainbowMiner** [comment]
Sorry, still not working:
```
     2m [2021-09-13 22:39:44.712] GPU0 Solution found in 1.30s (Submit every 82.68 seconds)
     2m [2021-09-13 22:39:46.301] Job 83b3ad
     2m [2021-09-13 22:39:48.515] Job 5d9aaa
     2m [2021-09-13 22:39:52.882] Job 4c7f12
     2m [2021-09-13 22:39:55.028] Job b6d153
     2m [2021-09-13 22:39:56.125] Job cb7367
     2m [2021-09-13 22:39:59.317] Job fb28be
     3m [2021-09-13 22:40:03.118] 
     3m [2021-09-13 22:40:03.118] 
     3m [2021-09-13 22:40:03.118] eu1.ethermine.org (ethash)    PING: 45ms    DIFFICULTY: 1.00     EPOCH: 440
     3m [2021-09-13 22:40:03.118]                                                                            
     3m [2021-09-13 22:40:03.118] 
     3m [2021-09-13 22:40:03.118] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     3m [2021-09-13 22:40:03.118] GPU6	3070	0/0	0	0	0	0	0.00	0.00
     3m [2021-09-13 22:40:03.118] 						0	0.00	0.00
     3m [2021-09-13 22:40:03.118] 
     3m [2021-09-13 22:40:03.118] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     3m [2021-09-13 22:40:03.118] GPU6	3070	Cuda	0.00 H/W	61.20 MH/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 22:40:03.118] 			0.00 H/s	61.20 MH/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 22:40:03.118]               
```

**Comment by sp-hash** [comment]
Try to create a copy of the folder, and let each instance run in its own folder.

**Comment by DeeDeeRanged** [comment]
Would it work with -U 0,6 ?

**Comment by sp-hash** [comment]
If two miners are running on the same computer, you need to run the miner in 2 separate folders. Did you try that?

**Comment by RainbowMiner** [comment]
> Did you try that?

Not yet - it's kind of difficult, since I use RainbowMiner for automation and start the miner(s). I have currently only one binary folder for each miner. I'll give it a try by running the miners manually.

**Comment by RainbowMiner** [comment]
If I start only the one miner instance on the one GPU6, the bug strikes, as well. So it has nothing to do with two instances running.

```
     2m [2021-09-14 09:54:08.387] 
     2m [2021-09-14 09:54:08.387] eu1.ethermine.org:4444 (ethash)    PING: 0ms    DIFFICULTY: 1.00     EPOCH: 440
     2m [2021-09-14 09:54:08.387]                                                                                
     2m [2021-09-14 09:54:08.387] 
     2m [2021-09-14 09:54:08.387] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     2m [2021-09-14 09:54:08.387] GPU6	3070	0/0	0	0	0	0	0.00	0.00
     2m [2021-09-14 09:54:08.387] 						0	0.00	0.00
     2m [2021-09-14 09:54:08.387] 
     2m [2021-09-14 09:54:08.387] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     2m [2021-09-14 09:54:08.387] GPU6	3070	Cuda	0.00 H/W	64.02 MH/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-14 09:54:08.387] 			0.00 H/s	64.02 MH/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-14 09:54:08.387]                                                                                
     2m [2021-09-14 09:54:08.387] 
     2m [2021-09-14 09:54:09.262] Job dd0011
```

**Comment by RainbowMiner** [comment]
> Would it work with -U [0,6] for devices?

Actually, a good idea! I tried:

#### --cuda-devices [0,1,2,3,4,5,6]

That works ok:
```
     1m [2021-09-14 12:50:58.674]
     1m [2021-09-14 12:50:58.674] ID    BOARD   TYPE    HASHRATE/W      HASHRATE        AVERAGE         SHARES
     1m [2021-09-14 12:50:58.674] GPU0  1070    Cuda    222.00 kH/W     29.30 MH/s      0.00 H/s        1/0/0 (100.00)
     1m [2021-09-14 12:50:58.674] GPU1  1070    Cuda    221.92 kH/W     28.85 MH/s      27.85 MH/s      3/0/1 (100.00)
     1m [2021-09-14 12:50:58.674] GPU2  1070    Cuda    209.60 kH/W     28.09 MH/s      0.00 H/s        1/0/0 (100.00)
     1m [2021-09-14 12:50:58.674] GPU3  1070    Cuda    204.75 kH/W     27.44 MH/s      27.01 MH/s      0/0/0 (100.00)
     1m [2021-09-14 12:50:58.674] GPU4  1070    Cuda    214.25 kH/W     28.50 MH/s      0.00 H/s        0/0/0 (100.00)
     1m [2021-09-14 12:50:58.674] GPU5  1070    Cuda    219.10 kH/W     28.48 MH/s      0.00 H/s        5/0/0 (100.00)
     1m [2021-09-14 12:50:58.674] GPU6  3070    Cuda    489.03 kH/W     64.06 MH/s      0.00 H/s        3/0/0 (100.00)
     1m [2021-09-14 12:50:58.674]                       1780.64 kH/s    234.72 MH/s     54.85 MH/s      13/0/1 (100.00)
     1m [2021-09-14 12:50:58.674]
     1m [2021-09-14 12:50:58.674]
```

#### --cuda-devices [0,6]

That doesn't work:
```
     2m [2021-09-14 12:53:30.163] ID    BOARD   TEMP    FAN     CORE    MEM     WATT    kW/h    COST/h
     2m [2021-09-14 12:53:30.163] GPU0  1070    62/0    46      2016    4404    132     0.13    inf
     2m [2021-09-14 12:53:30.163] GPU6  3070    0/0     0       0       0       0       0.00    0.00
     2m [2021-09-14 12:53:30.163]                                               132     0.13    inf
     2m [2021-09-14 12:53:30.163]
     2m [2021-09-14 12:53:30.163] ID    BOARD   TYPE    HASHRATE/W      HASHRATE        AVERAGE         SHARES
     2m [2021-09-14 12:53:30.163] GPU0  1070    Cuda    213.17 kH/W     28.14 MH/s      28.13 MH/s      2/0/0 (100.00)
     2m [2021-09-14 12:53:30.163] GPU6  3070    Cuda    0.00 H/W        64.20 MH/s      0.00 H/s        0/0/0 (100.00)
     2m [2021-09-14 12:53:30.163]                       213.17 kH/s     92.34 MH/s      28.13 MH/s      2/0/0 (100.00)
     2m [2021-09-14 12:53:30.163]
```

So it is clearly an indexing problem.

**Comment by sp-hash** [comment]
reproduced and fixed in v1.05

**Comment by RainbowMiner** [comment]
Confirmed. Solved. Thank you!
```
     2m [2021-09-17 11:42:42.645] eu1.ethermine.org (ethash)    PING: 57ms    DIFFICULTY: 1.00     EPOCH: 441
     2m [2021-09-17 11:42:42.645]                                                                            
     2m [2021-09-17 11:42:42.645] 
     2m [2021-09-17 11:42:42.645] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     2m [2021-09-17 11:42:42.645] GPU6	3070	63/0	46	1062	7992	131	0.13	inf
     2m [2021-09-17 11:42:42.645] 						131	0.13	inf
     2m [2021-09-17 11:42:42.645] 
     2m [2021-09-17 11:42:42.645] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     2m [2021-09-17 11:42:42.645] GPU6	3070	Cuda	487.19 kH/W	63.82 MH/s	63.69 MH/s	2/0/0 (100.00)
     2m [2021-09-17 11:42:42.645] 			487.19 kH/s	63.82 MH/s	63.69 MH/s	2/0/0 (100.00)
```
